packages/components/src/form-token-field/test/index.tsx

```diff
@@ -0,0 +1,2089 @@
+					__experimentalValidateInput={ startsWithCapitalLetter }
+				/>
+			);
+
+			// Add 'cranberry' token by typing it and pressing enter to tokenize it.
+			// The validation function won't allow the value from being tokenized.
+			// Note that the any token added before is still around, even if it
+			// wouldn't pass the newly added validation — this is because the
+			// validation happens when the input\'s value gets tokenized.
+			await user.type( input, 'cranberry[Enter]' );
+			expect( onChangeSpy ).toHaveBeenCalledTimes( 1 );
+			expectTokensToBeInTheDocument( [ 'cherry' ] );
+			expectTokensNotToBeInTheDocument( [ 'cranberry' ] );
+
+			// Retry, this time with capital letter. The value should be added.
+			await user.clear( input );
+			await user.type( input, 'Cranberry[Enter]' );
+			expect( onChangeSpy ).toHaveBeenCalledTimes( 2 );
+			expectTokensToBeInTheDocument( [ 'cherry', 'Cranberry' ] );
+		} );
+	} );
+
+	describe( 'maxLength', () => {
+		it( 'should not allow adding new tokens beyond the value defined by the `maxLength` prop', async () => {
+			const user = userEvent.setup( {
+				advanceTimers: jest.advanceTimersByTime,
+			} );
+
+			const onChangeSpy = jest.fn();
+
+			render(
+				<FormTokenFieldWithState
+					onChange={ onChangeSpy }
+					initialValue={ [ 'square', 'triangle', 'circle' ] }
+					maxLength={ 3 }
+				/>
+			);
+
+			expectTokensToBeInTheDocument( [ 'square', 'triangle', 'circle' ] );
+
+			const input = screen.getByRole( 'combobox' );
+
+			// Try to add the 'hexagon' token, but because the number of tokens already
+			// matches `maxLength`, the token won't be added.
+			await user.type( input, 'hexagon[Enter]' );
+			expect( onChangeSpy ).toHaveBeenCalledTimes( 0 );
+			expectTokensToBeInTheDocument( [ 'square', 'triangle', 'circle' ] );
+			expectTokensNotToBeInTheDocument( [ 'hexagon' ] );
+
+			// Delete the last token ("circle"), in order to make space for the
+			// hexagon token
+			await user.clear( input );
+			await user.keyboard( '[Backspace]' );
+			expect( onChangeSpy ).toHaveBeenCalledTimes( 1 );
+			expect( onChangeSpy ).toHaveBeenLastCalledWith( [
+				'square',
+				'triangle',
+			] );
+			expectTokensToBeInTheDocument( [ 'square', 'triangle' ] );
+			expectTokensNotToBeInTheDocument( [ 'circle' ] );
+
+			// Try to add the 'hexagon' token again. This time, the token will be
+			// added because the current number of tokens is below the `maxLength`
+			// threshold.
+			await user.type( input, 'hexagon[Enter]' );
+			expect( onChangeSpy ).toHaveBeenCalledTimes( 2 );
+			expect( onChangeSpy ).toHaveBeenLastCalledWith( [
+				'square',
+				'triangle',
+				'hexagon',
+			] );
+			expectTokensToBeInTheDocument( [
+				'square',
+				'triangle',
+				'hexagon',
+			] );
+		} );
+
+		// it( 'should not allow adding new tokens when the number of token already exceeds the value of the `maxLength` prop, while allowing the ones set before to stay as they are', () => {
```

Comment:
Was this one replaced by the above test and the below test? In that case, we could delete that line.